### PR TITLE
Build with go modules

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Build heketi using go modules. As one of the commit messages notes this is also a full dependencies refresh as a side effect.

The motivation to do this is largely due to go planning on [dropping support for non-modules builds in Go 1.17](https://blog.golang.org/go116-module-changes#TOC_2.) (the next release at the time of this post).  And we're concerned that we'll have to keep heketi on life support for some time after that release is out. If we need to make real changes or deal with security fixes in Go or dependencies after Go 1.17 is the only supported Go version we'd be having problems.

So it is.

### CODE REVIEW

**Review by iamniting — COMMENTED** [review]
these changes look good to me.

Moreover, we need some changes in our Docker file and GitHub action as well
https://github.com/heketi/heketi/blob/master/extras/docker/fromsource/Dockerfile
https://github.com/heketi/heketi/blob/master/.github/workflows/ci.yml

**Comment by phlogistonjohn** [comment]
I've marked this do not merge for now, despite taking it out of draft. It's basically ready for review but before we think about merging it it needs to be tested using the k8s executor, which our automated tests can't do.

So once I or someone (but let's face it, I probably need to do it myself) tests out these changes under k8s and things work OK we can then remove the label and think about merging it.

**Review by iamniting — APPROVED** [review]
Changes look good to me. 

There are some changes in Travis CI WRT the glide, But I think we can ignore them as of now as we are no longer considering Travis CI.

**Comment by phlogistonjohn** [comment]
Some basic testing on a k8s cluster has been performed. no obvious issues detected.

File: extras/release/Dockerfile
Comment (iamniting) [inline extras/release/Dockerfile:23]: Do not we want to run make commands here in this Dockerfile

**Review by iamniting — CHANGES_REQUESTED** [review]

**Comment by iamniting** [comment]
> Some basic testing on a k8s cluster has been performed. no obvious issues detected.

That's awesome

File: extras/release/Dockerfile
Comment (phlogistonjohn) [reply extras/release/Dockerfile:23]: Not for this. The idea here is the container merely acts as a "clean" environment for us to perform the builds. It doesn't do the work for us, unlike the fromsource container.

**Review by iamniting — APPROVED** [review]

**Review by raghavendra-talur — APPROVED** [review]

File: go.mod
Comment (pohly) [inline go.mod:1]: This has to include the version:
```
module github.com/heketi/heketi/v10
```

Then all imports in your code also need to use that instead of plain `github.com/heketi/heketi`.

It works as-is when working inside the repo, but any consumers referencing the code run into https://github.com/heketi/heketi/issues/1897

**Pull request merged**
